box-content-sdk/src/test/java/com/box/androidsdk/content/requests/BoxFolderRequestTest.java

```diff
@@ -50,4 +65,148 @@ public void testFolderUpdateRequest() throws NoSuchMethodException, BoxException
         String actual = updateReq.getStringBody();
         Assert.assertEquals(expected, actual);
     }
+
+    @Test
+    public void testFolderInfoRequest() throws Exception {
+        final String expectedRequestUrl = "https://api.box.com/2.0/folders/" + FOLDER_ID;
+
+        BoxApiFolder folderApi = new BoxApiFolder(SessionUtil.newMockBoxSession(mMockContext));
+        BoxRequestsFolder.GetFolderInfo folderInfo = folderApi.getInfoRequest(FOLDER_ID);
+
+        mockSuccessResponseWithJson(SAMPLE_FOLDER_JSON);
+        BoxFolder boxFolder = folderInfo.send();
+
+        Assert.assertEquals(expectedRequestUrl, folderInfo.mRequestUrlString);
```

Comment:
We should add a getter for `mRequestUrlString`

Reply:
@jimmy-holzer-box do you have any feedback on accessing protected fields directly inside test class in this manner?

Reply:
Depends :) If the field is made protected just to make it easier to access inside the test then no. If the filed is protected 'by design' then I think's it's ok to access it like this in the test.
That being said, I feel we have too many protected fields inside the app :)

Reply:
I agree, the member under discussion sounds like a good candidate to be package private IMO. Btw `@VisibleForTesting` is precisely used for this purpose. I have seen it being used in few sample apps. https://developer.android.com/reference/android/support/annotation/VisibleForTesting.html

Reply:
Think I'd lean towards using VisibleForTesting, if the test framework didn't allow for fields being accessed.